test/integration/mapping_validation_test.rb

```diff
@@ -20,6 +20,35 @@ class MappingValidationTest < ActiveSupport::TestCase
       #{invalid_categories}"
   end
 
+  test "every Shopify category has corresponding channel mappings" do
+    mappings_json_data = CLI.new.parse_json("dist/en/integrations/all_mappings.json")
+    shopify_category_lack_mappings = []
+
+    mappings_json_data["mappings"].each do |mapping|
+      next unless mapping["input_taxonomy"].include?("shopify")
+
+      all_shopify_category_ids = category_ids_from_taxonomy(mapping["input_taxonomy"])
+      next if all_shopify_category_ids.nil?
+
+      shopify_category_ids_from_mappings_input = Set.new
+      mapping["rules"].each do |rule|
+        shopify_category_ids_from_mappings_input.add(rule["input"]["category"]["id"])
+      end
+      shopify_category_ids_diff = all_shopify_category_ids - shopify_category_ids_from_mappings_input
```

Comment:
Should there be a return if this is empty?
```suggestion
      unmapped_category_ids = all_shopify_category_ids - shopify_category_ids_from_mappings_input
      next if unmapped_category_ids.empty?
```